libshaderc_spvc/src/spvc.cc

```diff
@@ -164,74 +174,370 @@ void shaderc_spvc_compiler_release(shaderc_spvc_compiler_t compiler) {
+      nullptr;  // NOT freed by our destructor
+
+  void consume_validation_message(spv_message_level_t level, const char* src,
+                                  const spv_position_t& pos,
+                                  const char* message) {
+    assert(result);
+    result->messages.append(message);
+    result->messages.append("\n");
+  }
+
+  static spv_result_t parse_header(void* user_data, spv_endianness_t endian,
+                                   uint32_t magic, uint32_t version,
+                                   uint32_t generator, uint32_t id_bound,
+                                   uint32_t reserved) {
+    return ((Compiler*)user_data)
+        ->parse_header(endian, magic, version, generator, id_bound, reserved);
+  }
+
+  spv_result_t parse_header(spv_endianness_t endian, uint32_t magic,
+                            uint32_t version, uint32_t generator,
+                            uint32_t id_bound, uint32_t reserved) {
+    ir.set_id_bounds(id_bound);
+    return SPV_SUCCESS;
+  }
+
+  static spv_result_t parse_instruction(void* user_data,
+                                        const spv_parsed_instruction_t* inst) {
+    return ((Compiler*)user_data)->parse_instruction(inst);
+  }
+
+  // Duplicate behavior of SPIRV-Cross.
+  spv_result_t parse_instruction(const spv_parsed_instruction_t* inst) {
+    switch (inst->opcode) {
+      case spv::OpExtInstImport: {
+        uint32_t id;
+        std::string ext;
+        GetOps(inst, id, ext);
+        if (ext == "GLSL.std.450")
+          set<spirv_cross::SPIRExtension>(id, spirv_cross::SPIRExtension::GLSL);
+        else if (ext == "SPV_AMD_shader_ballot")
+          set<spirv_cross::SPIRExtension>(
+              id, spirv_cross::SPIRExtension::SPV_AMD_shader_ballot);
+        else if (ext == "SPV_AMD_shader_explicit_vertex_parameter")
+          set<spirv_cross::SPIRExtension>(
+              id, spirv_cross::SPIRExtension::
+                      SPV_AMD_shader_explicit_vertex_parameter);
+        else if (ext == "SPV_AMD_shader_trinary_minmax")
+          set<spirv_cross::SPIRExtension>(
+              id, spirv_cross::SPIRExtension::SPV_AMD_shader_trinary_minmax);
+        else if (ext == "SPV_AMD_gcn_shader")
+          set<spirv_cross::SPIRExtension>(
+              id, spirv_cross::SPIRExtension::SPV_AMD_gcn_shader);
+        else
+          set<spirv_cross::SPIRExtension>(
+              id, spirv_cross::SPIRExtension::Unsupported);
+        // Other SPIR-V extensions which have ExtInstrs are currently not
+        // supported.
+      } break;
+
+      case spv::OpEntryPoint: {
+        spv::ExecutionModel model;
+        uint32_t id;
+        std::string name;
+        std::vector<uint32_t> interface_variables;
+        GetOps(inst, model, id, name, interface_variables);
+
+        auto itr = ir.entry_points.insert(
+            std::make_pair(id, spirv_cross::SPIREntryPoint(id, model, name)));
+        auto& e = itr.first->second;
```

Comment:
What is e here?   itr.first->second doesn't give me a clue at all.  Use a better name.  Having auto doesn't help either.